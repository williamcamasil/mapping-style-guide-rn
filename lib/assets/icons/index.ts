export * as Large from './large';
export * as Heavy from './heavy';
export * as Thin from './thin';
export * as Small from './small';
export * as Default from './default';
