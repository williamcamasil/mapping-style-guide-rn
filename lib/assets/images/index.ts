export { default as BtnTakePicture } from './BtnTakePicture';
export { default as BulletPoint } from './BulletPoint';
export { default as CheckboxImg } from './CheckboxImg';
export { default as DialogClose } from './DialogClose/index.svg';
export { default as Lock } from './Lock';
export { default as MaskedCardNumber } from './MaskedCardNumber';
export { default as Pointer } from './Pointer';
export { default as RadioImg } from './RadioImg';
