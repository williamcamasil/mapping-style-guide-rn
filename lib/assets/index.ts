export * as Icons from './icons';
export * as Images from './images';
export * as Lotties from './lotties';
