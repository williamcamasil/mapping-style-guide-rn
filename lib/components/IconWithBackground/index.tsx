import React from 'react';
import { ColorValue, StyleSheet, View } from 'react-native';

import { SvgProps } from 'react-native-svg';

import { useViewStyles } from '../../hooks';
import { AppThemeType, withTheme } from '../../theme';

export type IconWithBackgroundPropsType <IconProps extends SvgProps = SvgProps> = {
  theme: AppThemeType;
  Icon: React.FC<IconProps>;
  iconColor: ColorValue;
  iconSize: number;
  backgroundColor: ColorValue;
  backgroundSize: number;
};

const styles = StyleSheet.create({
  iconWrapper: {
    justifyContent: 'center',
    alignItems: 'center',
  },
});

const IconWithBackground: React.FC<IconWithBackgroundPropsType> = ({
  theme, Icon, iconColor, iconSize: size, backgroundColor, backgroundSize,
}) => {
  const iconWrapperStyle = useViewStyles(() => [
    styles.iconWrapper,
    {
      borderRadius: theme.borders.radius.medium,
      backgroundColor,
      width: backgroundSize,
      height: backgroundSize,
    },
  ], [theme, backgroundColor, backgroundSize]);

  return (
    <View style={iconWrapperStyle}>
      <Icon width={size} height={size} color={iconColor} />
    </View>
  );
};

export default withTheme(IconWithBackground);
