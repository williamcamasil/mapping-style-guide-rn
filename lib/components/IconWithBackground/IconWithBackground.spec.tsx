import React from 'react';

import {
  render,
} from '@testing-library/react-native';

import IconWithBackground from '.';
import { Icons } from '../../assets';
import { DefaultTheme } from '../../theme';

describe('IconWithBackground snapshot', () => {
  it('default', () => {
    const tree = render(
      <IconWithBackground
        Icon={Icons.Default.Airplane}
        iconColor={DefaultTheme.colors.primaryMain}
        iconSize={24}
        backgroundColor={DefaultTheme.colors.primary200}
        backgroundSize={36}
      />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
