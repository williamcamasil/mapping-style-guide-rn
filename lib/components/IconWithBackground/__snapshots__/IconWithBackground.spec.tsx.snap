// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconWithBackground snapshot default 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FDEBE3",
      "borderRadius": 12,
      "height": 36,
      "justifyContent": "center",
      "width": 36,
    }
  }
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight="24"
    bbWidth="24"
    fill="none"
    focusable={false}
    height={24}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 24,
          "width": 24,
        },
      ]
    }
    vbHeight={24}
    vbWidth={24}
    width={24}
  >
    <RNSVGGroup
      fill={null}
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGPath
        d="m14.081 7.116 3.312-3.312a1.982 1.982 0 0 1 2.803 2.803l-3.312 3.312 1.792 8.917c.066.328-.037.667-.273.904l-.66.66a1 1 0 0 1-1.639-.344l-2.624-6.733-3.406 3.406.182 2.17a1 1 0 0 1-.289.791l-1.086 1.086-2.12-3.536-3.537-2.121 1.086-1.086a.998.998 0 0 1 .791-.289l2.17.182 3.406-3.406-6.732-2.624A1 1 0 0 1 3.6 6.257l.66-.66a.999.999 0 0 1 .904-.273l8.917 1.792Z"
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
        propList={
          [
            "stroke",
            "strokeWidth",
            "strokeLinecap",
            "strokeLinejoin",
          ]
        }
        stroke={
          {
            "payload": 4293482516,
            "type": 0,
          }
        }
        strokeLinecap={1}
        strokeLinejoin={1}
        strokeWidth="1.8"
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;
